src/components/OpenWith/OpenWith.js

```diff
@@ -86,16 +102,96 @@ class OpenWith extends PureComponent<Props> {
     }
 
     /**
-     * Fetches the list of potential integrations on load
      *
      * @private
      * @inheritdoc
      * @return {void}
      */
     componentDidMount() {
-        /* no-op */
+        const { fileId }: Props = this.props;
+        this.api.getOpenWithAPI(false).get({
+            id: fileId,
+            successCallback: this.fetchAppIntegrations,
+            errorCallback: this.fetchErrorCallback
+        });
     }
 
+    /**
+     * Fetch app intgrations info needed to render.
+     *
+     * @param {OpenWithIntegrations} openWithintegrations - The available Open With integrations
+     * @return {void}
+     */
+    fetchAppIntegrations = (openWithintegrations: OpenWithIntegrations) => {
+        const { items } = openWithintegrations;
```

Comment:
This can happen in the function signature above.

Reply:
I don't think I can in this case because I use openWithIntegrations later. Unless there's some es6 de-structuring that I'm not aware of?